src/mds/MDSRank.cc

```diff
@@ -3781,6 +3784,29 @@ void MDSRankDispatcher::handle_conf_change(const ConfigProxy& conf, const std::s
   if (changed.count("mds_enable_op_tracker")) {
     op_tracker.set_tracking(conf->mds_enable_op_tracker);
   }
+  if (changed.count("mds_extraordinary_events_dump_interval")) {
+    reset_event_flags();
+    extraordinary_events_dump_interval =
+      conf.get_val<std::chrono::seconds>("mds_extraordinary_events_dump_interval").count();
+
+    std::string debug_mds = g_conf().get_val<std::string>("debug_mds");
+    std::stringstream ss(debug_mds.substr(debug_mds.find("/") + 1));
+    ss >> inmemory_log_level;
```

Comment:
I think you should use the `log_level` instead of the `gather_level` here ?

The format is `debug_mds = log_level/gather_level`.

Reply:
Another thing is that in product if the `gather_level` is `5` as default I am afraid the extraordinary events won't be collected in the logbuffer ? So you must set the `gather_level` at least to `10` in the code or let the users/customers do that manually ?

Reply:
> I think you should use the `log_level` instead of the `gather_level` here ?
> 
> The format is `debug_mds = log_level/gather_level`.

I was thinking I should check the `in-memory log level ` as mentioned in [1] ?
[1] https://docs.ceph.com/en/latest/rados/troubleshooting/log-and-debug/#ceph-subsystems

Reply:
I think we should check the `Log Level` instead. The `Memory Level` is only affect whether will the logs be buffered in memory.

The `g_ceph_context->_log->dump_recent();` will dump the recent events to the log files, and we should depend on the `Log Level` whether should dump the recent events to log files or not, if the `Log Level` already very high the log file should already have include those logs, no need to dump them again.

Reply:
Makes sense. 

We will also check if the `gather_level >= 10` before calling `g_ceph_context->_log->dump_recent()` and let the users set the `gather_level` manually which enables the in-memory log dump. Will update the long description also.

Reply:
Or we could just set the `gather_level` to `10` here when the `extraordinary_events_dump_interval` is enabled by users, no need to let users to set it separately. And just doc it in the option description and let the users know that ?

IMO we should make this as simple as possible.

Reply:
@lxbsz 
That's good idea. But what happens when the user disables `mds_extraordinary_events_dump_interval` in production? The `gather_level` still continues at `10`?

Reply:
IMO, that's should be okay just leave what it is currently, because users might have already changed it to `10` or others.

Reply:
> Or we could just set the `gather_level` to `10` here

More exactly this should be that if the `gather_level` is less than `10` we need to increase it to `10`, or no need to touch it.

> when the `extraordinary_events_dump_interval` is enabled by users, no need to let users to set it separately. And just doc it in the option description and let the users know that ?
> 
> IMO we should make this as simple as possible.